homeassistant/components/fritz/device_tracker.py

```diff
@@ -1,107 +1,185 @@
 """Support for FRITZ!Box routers."""
 import logging
+from typing import Dict
 
-from fritzconnection.core import exceptions as fritzexceptions
-from fritzconnection.lib.fritzhosts import FritzHosts
 import voluptuous as vol
 
-from homeassistant.components.device_tracker import (
-    DOMAIN,
-    PLATFORM_SCHEMA,
-    DeviceScanner,
-)
-from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_USERNAME
+from homeassistant.components.device_tracker import PLATFORM_SCHEMA, SOURCE_TYPE_ROUTER
+from homeassistant.components.device_tracker.config_entry import ScannerEntity
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
+from homeassistant.const import CONF_DEVICES, CONF_HOST, CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import callback
 import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.device_registry import CONNECTION_NETWORK_MAC
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.typing import ConfigType, HomeAssistantType
 
-_LOGGER = logging.getLogger(__name__)
+from .common import FritzBoxTools
+from .const import DEFAULT_DEVICE_NAME, DOMAIN
 
-DEFAULT_HOST = "169.254.1.1"  # This IP is valid for all FRITZ!Box routers.
-DEFAULT_USERNAME = "admin"
+_LOGGER = logging.getLogger(__name__)
 
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {
-        vol.Optional(CONF_HOST, default=DEFAULT_HOST): cv.string,
-        vol.Optional(CONF_USERNAME, default=DEFAULT_USERNAME): cv.string,
-        vol.Optional(CONF_PASSWORD): cv.string,
-    }
+YAML_DEFAULT_HOST = "169.254.1.1"
+YAML_DEFAULT_USERNAME = "admin"
+
+PLATFORM_SCHEMA = vol.All(
+    cv.deprecated(CONF_HOST),
+    cv.deprecated(CONF_USERNAME),
+    cv.deprecated(CONF_PASSWORD),
+    PLATFORM_SCHEMA.extend(
+        {
+            vol.Optional(CONF_HOST, default=YAML_DEFAULT_HOST): cv.string,
+            vol.Optional(CONF_USERNAME, default=YAML_DEFAULT_USERNAME): cv.string,
+            vol.Optional(CONF_PASSWORD): cv.string,
+        }
+    ),
 )
 
 
-def get_scanner(hass, config):
-    """Validate the configuration and return FritzBoxScanner."""
-    scanner = FritzBoxScanner(config[DOMAIN])
-    return scanner if scanner.success_init else None
+async def async_get_scanner(hass: HomeAssistantType, config: ConfigType):
+    """Import legacy FRITZ!Box configuration."""
+    if DOMAIN not in config:
+        return True
 
+    _LOGGER.debug("Import legacy FRITZ!Box configuration from YAML")
+    for entry_config in config[DOMAIN][CONF_DEVICES]:
+        hass.async_create_task(
```

Comment:
Since its being imported now, I think this means we can drop the breaking change

Reply:
Yup, removed